addons/website_sale/models/sale_order_line.py

```diff
@@ -102,8 +102,13 @@ def _get_cart_display_price(self):
     def _check_validity(self):
         if (
             not self.combo_item_id
-            and sum(self._get_lines_with_price().mapped('price_unit')) == 0
-            and self.order_id.website_id.prevent_zero_price_sale
+            and (
+                self.order_id.website_id.prevent_sale
+                and self.order_id.website_id._prevent_product_sale(
+                    self.product_template_id,
+                    sum(self._get_lines_with_price().mapped('price_unit')) == 0
+                )
+            )
```

Comment:
```suggestion
            and self.order_id.website_id.prevent_sale
            and self.order_id.website_id._prevent_product_sale(
                self.product_template_id,
                sum(self._get_lines_with_price().mapped('price_unit')) == 0
            )
```
No need for extra bracket 🙈